libs/components/datetime/src/lib/modules/datepicker/datepicker.component.ts

```diff
@@ -429,4 +437,23 @@ export class SkyDatepickerComponent implements OnDestroy, OnInit {
       this.#customDatesSubscription = undefined;
     }
   }
+
+  #populateInputBox(): void {
+    const hintTextObs: Observable<string | undefined> = this.dateFormat
+      ? this.#resourceSvc.getString(
+          'skyux_datepicker_format_hint_text',
+          this.dateFormat,
+        )
+      : of('');
+
+    hintTextObs.pipe(take(1)).subscribe((formatString) => {
+      if (this.inputBoxHostService && this.inputTemplateRef) {
+        this.inputBoxHostService.populate({
+          inputTemplate: this.inputTemplateRef,
+          buttonsTemplate: this.triggerButtonTemplateRef,
+          hintText: formatString,
```

Comment:
Up until now, the `SkyInputBoxHostService.populate` method has been used exclusively for templates. Since this is a property whose value can change whenever, it might be useful to have a different pattern to handle those types of things. Maybe a new public `SkyInputBoxHostService.setHintText` method? Another reason for doing it that way: we already have a `hintTextTemplate` property internal to input box and I can imagine needing that added to the `populate` method in the future.

Also, it seems that [the original intent](https://github.com/blackbaud/skyux/blob/main/libs/components/forms/src/lib/modules/input-box/input-box-host.service.ts#L37) of the `populate` method was to be called within an Angular lifecycle hook (so tying it to a dynamic input might be an issue if it gets called every time the date format changes).